File: frontend/types/portfolio/project-link.ts
import type {
  ProjectLinkType,
} from "@/types/portfolio/common"

export type ProjectLinkRead = {
  id: string
  project_id: string

  label: string
  url: string

  link_type: ProjectLinkType

  icon: string | null

  opens_in_new_tab: boolean

  sort_order: number
}

File: mobile/shared/src/commonMain/kotlin/com/akahalu/portfolio/domain/portfolio/model/ProjectLink.kt
package com.akahalu.portfolio.domain.portfolio.model

enum class ProjectLinkType {
    REPOSITORY,
    LIVE_DEMO,
    DOCUMENTATION,
    CASE_STUDY,
    VIDEO,
    DOWNLOAD,
    APP_STORE,
    PLAY_STORE,
    DESIGN,
    ARTICLE,
    API,
    OTHER,
}

data class ProjectLink(
    val id: String,
    val projectId: String,
    val label: String,
    val url: String,
    val linkType: ProjectLinkType,
    val icon: String?,
    val opensInNewTab: Boolean,
    val sortOrder: Int,
)